--- main2.c	2025-11-23 23:13:42.750270837 +0300
+++ main3.c	2025-11-23 23:17:29.926730811 +0300
@@ -40,26 +40,30 @@
 
 int main(int argc, char* argv[]) {
 
-    if (argc != 3) {
+    if (argc != 4) {
         fprintf(stderr, "Error: incorrect argc (argc = %d)\n", argc);
         return 1;
     }
 
-    if (strlen(argv[1]) != 2) {
+    int seed = atoi(argv[1]);
+    if (seed < 0) {
+        fprintf(stderr, "Error: incorrect seed (expect positive seed)\n");
+        return 1;
+    }
+
+    if (strlen(argv[2]) != 2) {
         fprintf(stderr, "Error: incorrect argv[1] length (%lu)\n", strlen(argv[1]));
         return 1;
     }
-    char space = argv[1][0];
-    char wall = argv[1][1];
+    char space = argv[2][0];
+    char wall = argv[2][1];
 
-    int num_rooms = atoi(argv[2]);
+    int num_rooms = atoi(argv[3]);
     if (num_rooms <= 0) {
         fprintf(stderr, "Error: incorrect number of rooms\n");
         return 1;
     }
 
-    int seed = 42;
-
     srand(seed);
 
     int size = num_rooms * 2 + 1;
